### PR TITLE
Update ThirdPartyNoticeText.txt

### LINKED ISSUE #52778: Broken Unicode in ThirdPartyNoticeText.txt
# Bug Report

Seems that `ThirdPartyNoticeText.txt` has some bad unicode characters, at least python 3 fails to parse it 😅 

```shell
$ cd /tmp
$ git clone --depth 1 https://github.com/microsoft/typescript
$ python3.11
>>> with open('typescript/ThirdPartyNoticeText.txt') as file_handle:
...     for line in file_handle.readlines():
...         print()
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<frozen codecs>", line 322, in decode
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa0 in position 6038: invalid start byte
```

I could provide a PR to fix it, if such a PR is welcomed 😄

### PR DESCRIPTION
**Pull request by gforcada**
Fixes #52778

### CODE REVIEW

**Comment by gforcada** [comment]
@microsoft-github-policy-service agree

**Comment by fatcerberus** [comment]
What's up with this?

image

Are we *certain* the Unicode in the file was actually broken before?

File: ThirdPartyNoticeText.txt [hunk 1]
```diff
@@ -85,12 +85,12 @@ DOM
...
-Permission to copy, modify, and distribute this work, with or without modification,�for any purpose and without fee or royalty is hereby granted, provided that you include the following 
+Permission to copy, modify, and distribute this work, with or without modification, for any purpose and without fee or royalty is hereby granted, provided that you include the following
 on ALL copies of the work or portions thereof, including modifications:
 * The full text of this NOTICE in a location viewable to users of the redistributed or derivative work.
 * Any pre-existing intellectual property disclaimers, notices, or terms and conditions. If none exist, the W3C Software and Document Short Notice should be included.
 * Notice of any changes or modifications, through a copyright statement on the new code or document such as "This software or document includes material copied from or derived 
-from [title and URI of the W3C document]. Copyright � [YEAR] W3C� (MIT, ERCIM, Keio, Beihang)." 
+from [title and URI of the W3C document]. Copyright (C) [YEAR] W3C(C) (MIT, ERCIM, Keio, Beihang)."
```
Comment (fatcerberus) [inline ThirdPartyNoticeText.txt:93]: That second one is an `(R)` (restricted trademark), not a `(C)` (copyright).

Comment (gforcada) [reply ThirdPartyNoticeText.txt:93]: Oh right, as I was replacing it for the third time I was not seeing it properly, thanks, for spotting it out 🔍

**Comment by gforcada** [comment]
> What's up with this?
> 
> image
> 
> Are we _certain_ the Unicode in the file was actually broken before? Because this seems to have made it worse... eyes

That's interesting, indeed, seems it is now worse than before somehow, I copied these lines from the actual website, and my editor was showing them fine, but when I copied the text from my editor to GitHub to create this PR something might have gone wrong 😕

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
Please make this valid UTF-8 rather than trying to turn it all into ASCII

File: ThirdPartyNoticeText.txt (hunk 1 above)
Comment (DanielRosenwasser) [reply ThirdPartyNoticeText.txt:93]: What's the problem with these being Unicode symbols?

File: ThirdPartyNoticeText.txt [hunk 2]
```diff
@@ -136,22 +136,22 @@ Any other capitalized terms not specifically defined herein have the same meanin
...
...ed to grant the rights and promises set forth in this Agreement. IN ALL OTHER RESPECTS THE SPECIFICATION IS PROVIDED “AS IS.” The entire risk as to implementing or otherwise using the Specification is assumed by the implementer and user. Except as stated herein, I expressly disclaim any warranties (express, implied, or otherwise), including implied warranties of merchantability, non-infringement, fitness for a particular purpose, or title, related to the Specification. IN NO EVENT WILL ANY PARTY BE LIABLE TO ANY OTHER PARTY FOR LOST PROFITS OR ANY FORM OF INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES OF ANY CHARACTER FROM ANY CAUSES OF ACTION OF ANY KIND WITH RESPECT TO THIS AGREEMENT, WHETHER BASED ON BREACH OF CONTRACT, TORT (INCLUDING NEGLIGENCE), OR OTHERWISE, AND WHETHER OR NOT THE OTHER PARTY HAS BEEN ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. All of my obligations under Section 3 regarding the transfer, successors in interest, or assignment of Granted Claims will be satisfied if I notify the transferee or assignee of any patent that I know contains Granted Claims of the obligations under Section 3. Nothing in this Agreement requires me to undertake a patent search.
```
Comment (DanielRosenwasser) [inline ThirdPartyNoticeText.txt:139]: > â€œAS IS.â€ 

This is worse, right?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
the file is now UTF-8, but the characters are wrong. Seems like you need to fix up the invalid characters with their UTF-8 equivalents.

**Comment by DanielRosenwasser** [comment]
Sorry - I didn't see the previous comments from Ryan.

**Comment by DanielRosenwasser** [comment]
I think what happened is that each of these special characters is UTF16 - but the rest of the file is actually UTF8/ASCII.

**Comment by gforcada** [comment]
Indeed, that's the point:

```shell
$ file -i ThirdPartyNoticeText.txt 
ThirdPartyNoticeText.txt: text/plain; charset=unknown-8bit
```

Due to these few characters that are in `UTF-16` the encoding of the file is not consistent, `iconv` fails to convert it to `UTF-8` as well 😕 

I will try to extract the characters alone, convert them and add them back 🤞🏾 🍀

**Comment by jakebailey** [comment]
Here are some steps that work to fix the file for me:

1. Open the file in VS Code.
2. At the bottom right, click "UTF-8".
3. Select the detected encoding, which is some "Western (Windows 1252)" encoding.
4. At the bottom right, click "Windows 1252".
5. Save as "UTF-8".

**Comment by RyanCavanaugh** [comment]
Closing since it's been a couple weeks since changes were requested. New PR meeting the previously-given criteria are welcome.

**Pull request closed without merging**